software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -395,12 +390,12 @@ public void joinCluster(String nodeName) {
             if (!hasJoinedCluster()) {
                 ScriptHelper joinClusterScript = newScript("joinCluster")
                         .body.append(sudo(format("%s cluster join %s", getRiakAdminCmd(), nodeName)))
+                        .body.append(sudo(format("%s cluster plan", getRiakAdminCmd())))
+                        .body.append(sudo(format("%s cluster commit", getRiakAdminCmd())))
```

Comment:
Actually I found why it is implemented that way.
I think the initial idea was to support cluster queue operations. It is correct not to commit just after the cluster join command. Other than that it was executed for one of the members.
Instead of that commit was executed after the joinings are done and then in the RiakCluster.start at the end it was executed clusterCommit()
The bug here is that the commit is not executed properly because every time joining was executed against random member of the ring instead of using one, so commiting should be done against the nodes or the node which triggered cluster join. This is the actual bug. And the other bug is may be that leaveCluster had plan and commit command which is unnecessary.

My point is that it could support queue cluster operations but the task requires bigger effort.